File: backend/src/main/java/com/bookbook/domain/wishList/repository/WishListRepository.java
package com.bookbook.domain.wishList.repository;

import com.bookbook.domain.wishList.entity.WishList;
import com.bookbook.domain.wishList.enums.WishListStatus;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface WishListRepository extends JpaRepository<WishList, Integer> {

    // Status를 포함한 조회 메서드들
    List<WishList> findByUserIdAndStatusOrderByCreatedDateDesc(Long userId, WishListStatus status);

    Optional<WishList> findByUserIdAndRentIdAndStatus(Long userId, Integer rentId, WishListStatus status);
}

File: backend/src/main/java/com/bookbook/domain/wishList/repository/WishListRepository.kt
package com.bookbook.domain.wishList.repository

import com.bookbook.domain.wishList.entity.WishList
import com.bookbook.domain.wishList.enums.WishListStatus
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface WishListRepository : JpaRepository<WishList, Int> {
    fun findByUserIdAndStatusOrderByCreatedDateDesc(userId: Long, status: WishListStatus): List<WishList>

    fun findByUserIdAndRentIdAndStatus(userId: Long, rentId: Int, status: WishListStatus): WishList?
}
